miner/worker.go

```diff
@@ -1153,7 +1154,13 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 		}
 		header.Coinbase = w.coinbase
 		if useMB {
-			header.Coinbase = maybeMB.Coinbase
+			if n := uint64(time.Now().Unix()); n < maybeMB.Timestamp {
```

Comment:
I'm not sure I like this check. It seems very susceptible to minor clock drift. I'd think I'd prefer something like https://github.com/flashbots/mev-geth/blob/master/miner/worker.go#L1041, where it overwrites the timestamp only if it's invalid (or maybe reject)